tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas3_matrix_product_kk.hpp

```diff
@@ -254,22 +254,28 @@ void trmm_left(KokkosExecSpace &&exec, Triangle t, DiagonalStorage d,
       std::experimental::linalg::lower_triangle_t>;
   constexpr bool explicit_diag = std::is_same_v<DiagonalStorage,
       std::experimental::linalg::explicit_diagonal_t>;
-  const auto C_ext0 = C_view.extent(0);
+  const auto C_ext0 = C_view.extent(0); // == A_ext0 == A_ext1
+  const auto C_ext1 = C_view.extent(1);
 
-  KokkosKernelsSTD::Impl::ParallelMatrixVisitor v(std::move(exec), C_view);
-  v.for_each_matrix_element(
-    KOKKOS_LAMBDA(const auto i, const auto j) {
-      decltype(auto) cij = C_view(i, j);
-      cij = c_element_type{};
-      const size_type k0 = lower ? (explicit_diag ? i : i + 1) : 0;
-      const size_type k1 = lower ? C_ext0 :(explicit_diag ? i + 1 : i);
-      for (size_type k = k0; k < k1; ++k) {
-        cij += A_view(i, k) * B_view(k, j);
-      }
-      if constexpr (!explicit_diag) {
-        C_view(i, j) += /* 1 times */ B_view(i, j);
-      }
-    });
+  // Note: This routine can be called in-place with B_view=C_view
+  // because (i,j) indexing respects C updating order
+  // and parallelism is restricted accordingly
+  for (size_type ii = 0; ii < C_ext0; ++ii) {
```

Comment:
Did y'all ever find it weird that the "kokkos-kernels" back end can't actually call kokkos-kernels for most of the algorithms, because kokkos-kernels never bothered to wrap most of the BLAS?